src/DynamoCoreWpf/Utilities/ActionDebouncer.cs

```diff
@@ -36,6 +36,11 @@ public void Debounce(int timeout, Action action)
             Cancel();
             cts = new CancellationTokenSource();
 
+            // The TaskScheduler.FromCurrentSynchronizationContext() exists only if there is a valid SyncronizationContex.
+            // Calling this method from a non UI thread could have a null SyncronizationContex.Current,
+            // so in that case we use the default TaskScheduler which uses the thread pool.
+            var taskScheduler = SynchronizationContext.Current != null ? TaskScheduler.FromCurrentSynchronizationContext() : TaskScheduler.Default;
```

Comment:
This change makes it a bit harder to predict where the continuation will be executed. The alternative is to identify all tests that call into the ActionDebouncer and make sure they setup a SyncContext before (seemed too tedious to do)

Reply:
if the sync context is not on the UI thread can we log something?

Reply:
The sync context should always be setup in a UI thread. But I can add it if it makes it more understandable